src/cargo/ops/cargo_doc.rs

```diff
@@ -61,38 +67,47 @@ pub fn doc(manifest_path: &Path,
 }
 
 #[cfg(not(any(target_os = "windows", target_os = "macos")))]
-fn open_docs(path: &Path) {
+fn open_docs(path: &Path) -> Result<(), ()> {
+    // trying $BROWSER
+    match env::var("BROWSER").map(|name| Command::new(name).arg(path).status()) {
```

Comment:
It's insofar a 'standard' that it is used somewhat widely.

https://wiki.archlinux.org/index.php/environment_variables
https://unix.stackexchange.com/questions/213367/where-do-editor-pager-browser-environment-variables-come-from